src/os/bluestore/BlueStore.cc

```diff
@@ -6230,29 +6236,40 @@ void BlueStore::_txc_state_proc(TransContext *txc)
 	  // sequencer that is committing serially it is possible to keep
 	  // submitting new transactions fast enough that we get stuck doing
 	  // so.  the alternative is to block here... fixme?
+	} else if (txc->osr->txc_with_unstable_io) {
+	  dout(20) << __func__ << " prior txc(s) with unstable ios "
+		   << txc->osr->txc_with_unstable_io.load() << dendl;
 	} else if (g_conf->bluestore_debug_randomize_serial_transaction &&
 		   rand() % g_conf->bluestore_debug_randomize_serial_transaction
 		   == 0) {
 	  dout(20) << __func__ << " DEBUG randomly forcing submit via kv thread"
 		   << dendl;
 	} else {
 	  _txc_finalize_kv(txc, txc->t);
-	  txc->kv_submitted = true;
+	  txc->state = TransContext::STATE_KV_SUBMITTED;
+	  if (g_conf->bluestore_rtc &&
+	      txc->osr->kv_finisher_submitting == 1) {
```

Comment:
@voidbag, i couldn't find the symbol in the PR, so got lost where it is getting modified. Thanks for the explanation. Will look at the other PR also.